lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -603,6 +603,120 @@ SILCombiner::optimizeConcatenationOfStringLiterals(ApplyInst *AI) {
   return tryToConcatenateStrings(AI, Builder);
 }
 
+/// This routine replaces the old witness method inst with a new one.
+void SILCombiner::replaceWitnessMethodInst(
+    FullApplySite Apply, WitnessMethodInst *WMI, SILBuilderContext &BuilderCtx,
+    const CanType &ConcreteType, const ProtocolConformanceRef ConformanceRef) {
+  SILBuilderWithScope WMIBuilder(WMI, BuilderCtx);
+  auto *NewWMI = WMIBuilder.createWitnessMethod(
+      WMI->getLoc(), ConcreteType, ConformanceRef, WMI->getMember(),
+      WMI->getType());
+  MutableArrayRef<Operand> Operands = Apply.getInstruction()->getAllOperands();
+  for (auto &Op : Operands) {
+    if (Op.get() == WMI)
+      Op.set(NewWMI);
+  }
+  if (WMI->use_empty())
+    eraseInstFromFunction(*WMI);
```

Comment:
If it has other uses, they could also be replaced with NewWMI. In fact you could avoid passing in the Apply here, and look at all uses of the WMI instead.